### ISSUE TITLE
immutable-data exception for class properties

### ISSUE THREAD

**Issue by themaxhero** [jonaskello/eslint-plugin-functional#159]
I want to use Immutability all over the project in my work.
But there is a problem.
Our project is made in Angular, and It's not that trivial to make it work without classes and class properties mutation.

Is there any plan to include this exception?
If not, I would like to know if I implement this, It would be merged?

Thank you for your attention. Have a nice day.

**Comment by RebeccaStevens** [comment]
Would you be able to provide a code example of what you want to be able to do?

**Comment by themaxhero** [comment]
> Would you be able to provide a code example of what you want to be able to do?

example:
```ts
@Component({
  selector: "app-club",
  templateUrl: "./club.component.html",
  styleUrls: ["./club.component.scss"],
})
export class ClubComponent implements OnInit {
  someProperty: string[] = [];
  
  ngOnInit(): void {
    let someVar = 1;
    someVar = 300; // <--- this should raise an error or warning based on configuration
    this.someProperty = ["a", "b", "c"]; // this should be accepted by setting ignoreClassProperty to true
  }
}
```

**Comment by RebeccaStevens** [comment]
You should be able to just use the "ignoreAccessorPattern" to achieve this:

```jsonc
{
  "functional/immutable-data": [
    "error",
    {
      "ignoreAccessorPattern": "this.*" // or "this.**" to allow deep mutations
    }
  ]
}
```

Does this achieve what you are wanting?

**Comment by themaxhero** [comment]
Oh, thank you. :)
